www/index.js

```diff
@@ -47,14 +47,21 @@ const canvas = document.createElement('canvas');
 canvas.id = 'canvas';
 document.body.appendChild(canvas);
 
-const game = wasm.make_game()
+const maybeGame = wasm.make_game();
+if (maybeGame.get_error()) {
+  throw new Error(maybeGame.get_error());
+}
+const game = maybeGame.get_game();
+if (!game) {
+  throw new Error(`Failed to make a Game object`);
+}
 function drawOneFrame() {
-  const result = game.draw(
+  const maybeError = game.draw(
       curInput.up, curInput.down, curInput.left, curInput.right);
```

Comment:
An interesting detail to consider here is that meaningful input changes may occur in the stretch of frames between two `rAF` callbacks. For example, the `e` key pressed and released in the space of a frame could have meaningful consequences. Add to that: `rAF` doesn't run at the same frequency on all devices now that we have 90hz+ displays...

Anyway, all that to say that we may want to consider some way to distinguish game tick rate from draw rate.

Reply:
+1! On it